### PR TITLE
Tlg0006.tlg009

### LINKED ISSUE #130: (tlg0006.tlg009) epidoc/capitains of Euripides Heracles

### PR DESCRIPTION
**Pull request by balmas**
Fixes #130 

@lcerrato can you please look this over for me? This is one of the texts that was in the Perseids annotsrc inventory, that we had partially converted before the CapiTaiNs guidelines so that it could be used in Perseids. Anise Ferreira has a student who was using it and is missing it since I switched over to the CTS 5 environment on Nemo and Nautilus for annotation sources in Perseids.

This is a really tricky text because the line numbers in the source text are sloppy and it required a fair amount of editorial judgement on my part as to how to make it suitable for canonical citation (i.e. no duplicates).  I have no idea if I'm making horrible scholarly errors here though...

Thanks!

### CODE REVIEW

File: data/tlg0006/tlg009/tlg0006.tlg009.perseus-grc2.xml [hunk 1]
```diff
@@ -2195,9 +2195,9 @@
                             <l n="1170">σῴσας με νέρθεν, ἦλθον, εἴ τι δεῖ, γέρον,</l>
                             <l n="1171">ἢ χειρὸς ὑμᾶς τῆς ἐμῆς ἢ συμμάχων.</l>
                             <l n="1172">ἔα·</l>
-                            <l n="1173">τί νεκρῶν τῶνδε πληθύει πέδον;</l>
-                            <l n="1174">οὔ που λέλειμμαι καὶ νεωτέρων κακῶν</l>
-                            <l n="1175">ὕστερος ἀφῖγμαι; τίς τάδʼ ἔκτεινεν τέκνα;</l>
+                            <l n="1172a">τί νεκρῶν τῶνδε πληθύει πέδον;</l>
```
Comment (lcerrato) [inline data/tlg0006/tlg009/tlg0006.tlg009.perseus-grc2.xml]: Hi, @balmas In this instance, line 1172 starts with ἔα and ends with πέδον.
It's just that there is an indentation in the page formatting.
I haven't worked on much drama, so I don't know the best solution to this offhand.

**Comment by lcerrato** [comment]
Hi, @balmas,

We had
`<l n="1172">ἔα·</l>`
`<l n="1173">τί νεκρῶν τῶνδε πληθύει πέδον;</l>`

and you are proposing 
`<l n="1172">ἔα·</l>`
` <l n="1172a">τί νεκρῶν τῶνδε πληθύει πέδον;</l>`

Which is how we'd want this instance to read.

@TDBuck has been dealing with something similar in the drama he has worked on, but I don't think we have any documentation on this as yet.
It involves using ` <lb n=(\d+)> ` in conjunction with `<l n=(\d+)>`
I think in the examples of his approach I have seen so far, this combines the automatic line numbers (the sequential through numbers usually added via transform) with an effort to preserve the print based numbers. It is fairly time consuming but might be useful.

I can go ahead and see if I can replicate that in this case, and ask for further input.

The refsDecl still uses `<l>` not `<lb>` and the format is 
(\d+) 
(\d+)a 
(\d+)ab 
(\d+)ac 
(\d+)ad
etc.

**Comment by balmas** [comment]
Ah, yes now I remember that discussion. I think that using the lb tags is a good idea, but in that case, I think I would actually do the following, e.g.

<l n="1172">ἔα·<lb/>
τί νεκρῶν τῶνδε πληθύει πέδον;</l>

Thereby preserving both the line break, and the "canonical" line numbering.  I think that's actually a better solution than what I did, with adding the 'a' but the problem I have with doing it is that, given that I don't read Greek, it's not always clear to me if I've interpreted the indentation properly. There's also the question of whether we should redo the whole text this way, or just for the areas which are problematic. 

Maybe we should ask Anise and her student if they want to tackle this as part of their work with the text??

**Comment by lcerrato** [comment]
Yes, this an elegant solution and I'd prefer this (where applicable).
@TDBuck points out this does not work when the break spans two speakers. 

I don't mind reviewing the whole play. It shouldn't be too hard to spot the areas where it goes out of sync. 
Then the end users can review that pass.

(And, I guess none of this matters for CTS, since for CTS, these are arbitrary labels.)

**Comment by balmas** [comment]
I think for the case where the break spans speakers we can switch to using the `<said/>` tags rather than the `<speaker/>` blocks. Take a look at what I did in 002478b96c9f08d88c4594f05e248440f17653b1 and let me know if you think that could work.

**Comment by TDBuck** [comment]
Does alternating between <said/> and <sp/> and <speaker/> cause problems with CTS?

**Comment by lcerrato** [comment]
I really think we need some up to date documentation for drama. I'm not up to date on the various adaptations that are being suggested for all of these points.

**Comment by TDBuck** [comment]
@balmas I agree that we shouldn't alternate. This sounds like a better fix than what I was doing then. I can start converting the Aristophanes I was working on.

**Comment by balmas** [comment]
ok, so perhaps I should do that here, convert all the speakers to said tags, and then go back through and check all the line numbers with alphas to see what needs to be fixed.  @lcerrato if I did that, would you be able to double-check my work?

**Comment by lcerrato** [comment]
Of course, will review as needed.
Thank you both for clarifying the speaker to said tags issue. I knew about the proposal, but had not seen it consistently implemented.
I believe I may have previously submitted work that will require revision.

**Comment by balmas** [comment]
Ok, following up on our conversation yesterday:

1. if we use said tags, the entries in the `@who` attribute need to be an idref, pointing at the xml:id of previously identified entity (e.g. `<said who="#antigone">...</said>` where there is somewhere in the document a `<person xml:id="antigone">...</person>`
2. epidoc doesn't support `<castList>` which would normally be where we put these persons, as `<castItem/>` elements, and we have previously decided that we would use `<listPerson/>` in the `teiHeader/sourceDesc` element.  (see e..g https://github.com/PerseusDL/canonical-latinLit/blob/5d75f4277dca1289a5fd21b8cf36172feb6239bc/data/phi1017/phi003/phi1017.phi003.perseus-lat2.xml)
3. many of the primary source texts for drama didn't include a castList element to begin with, but in replacing the speaker/sp construct with said tags we need also to add these elements (now as listPerson/person in the header) so that the references in the who attributes are pointing at valid elements
4. if a castList did exist, it needs to be converted.
5. @TDBuck is going to investigate whether we can use greek unicode chars in our xml:id values. if not, we should use the english equivalent on the greek character name as the xml:id and supply the greek in the `persName` child element of the `person` element
6. @TDBuck can you also write all this up and add it to the tei-conversion-tools wiki?

**Comment by balmas** [comment]
ok, @lcerrato I'm ready for you to take a look at this again. All the speaker tags have been converted to said. What needs to be done now is review all the line numbers which have alphas (and their surrounding lines), and fix them up as compared to the original text, merging with the `<l/>` tags and adding  '<lb/>' tags where necessary. Here's the list of all the alpha references:

107a
122b
216a
353a
369a
410b
427b
459a
514a
629b
755a
757a
765a
771b
772a
772b
791b
808a
894a
894b
900a
910a
912a
914a
914b
915a
1049a
1049b
1051a
1051b
1052a
1052b
1054a
1054b
1067a
1068a
1068b
1069a
1069b
1070a
1070b
1172a
1199a
1199b
1396a

**Comment by lcerrato** [comment]
Ok, I have reviewed all of the disputed lines, and made changes where appropriate.

**Comment by balmas** [comment]
Thanks @lcerrato ! Merging!

**Pull request merged**